Sources/StreamChatUI/Utils/ChatChannelNamer.swift

```diff
@@ -5,24 +5,28 @@
 import Foundation
 import StreamChat
 
-open class ChatChannelNamer {
-    open var maxMemberNames: Int { 2 }
-    open var separator: String { "," }
-    
-    public required init() {}
-    
-    /// Generates a name for the given channel, given the current user's id.
-    ///
-    /// The priority order is:
-    /// - Assigned name of the channel, if not empty
-    /// - If the channel is direct message (implicit cid):
-    ///   - Name generated from cached members of the channel
-    /// - Channel's id
-    /// - Parameters:
-    ///   - channel: Channel to generate name for.
-    ///   - currentUserId: Logged-in user. This parameter is used when deciding which member's names are going to be displayed.
-    /// - Returns: A valid Channel name.
-    open func name<ExtraData: ExtraDataTypes>(for channel: _ChatChannel<ExtraData>, as currentUserId: UserId?) -> String {
+public typealias ChatChannelNamer = _ChatChannelNamer
+public typealias _ChatChannelNamer<ExtraData: ExtraDataTypes> =
+    (_ channel: _ChatChannel<ExtraData>, _ currentUserId: UserId?) -> String
+
+/// Generates a name for the given channel, given the current user's id.
+///
+/// The priority order is:
+/// - Assigned name of the channel, if not empty
+/// - If the channel is direct message (implicit cid):
+///   - Name generated from cached members of the channel
+/// - Channel's id
+///
+/// - Parameters:
+///   - maxMemberNames: Maximum number of visible members in Channel defaults to `2`
+///   - separator: Separator of the members, defaults to `y`
+/// - Returns: A closure with 2 parameters carrying `channel` used for name generation and `currentUserId` to decide
+/// which members' names are going to be displayed
+public func DefaultChatChannelNamer<ExtraData: ExtraDataTypes>(
+    maxMemberNames: Int = 2,
+    separator: String = ","
+) -> _ChatChannelNamer<ExtraData> {
```

Comment:
This is very detailed documentation but IMO the main point is missing :) WHAT does the generated name actually look like? Can you add examples for various cases, please?

Reply:
Added :)